src/opening-period/page/CreateNewOpeningPeriodPage.tsx

```diff
@@ -73,30 +120,46 @@ export default function CreateNewOpeningPeriodPage({
           parse(data.openingPeriodEndDate, dateFormFormat, new Date()),
           dateApiFormat
         ),
-        resource_state: ResourceState.OPEN,
         override: false,
-        time_span_groups: [],
+        time_span_groups: [
+          {
+            time_spans: formatTimeSpansToApiFormat(data.timeSpans),
+            rules: [],
+          },
+        ],
       });
       setSubmitStatus('succeeded');
     } catch (err) {
       setSubmitStatus('error');
+      // eslint-disable-next-line no-console
+      console.error(err); // For debug purposes
     }
   };
 
   useEffect((): void => {
     // UseEffect's callbacks are synchronous to prevent a race condition.
     // We can not use an async function as an useEffect's callback because it would return Promise<void>
-    api
-      .getResource(resourceId)
-      .then((r: Resource) => {
-        setResource(r);
+
+    Promise.all([api.getResource(resourceId), api.getDatePeriodFormOptions()])
+      .then((values) => {
+        setResource(values[0] as Resource);
+        const resourceStateOptionsInApiFormat =
+          values[1].actions.POST.resource_state.choices;
+        setResourceStateOptions(
+          resourceStateOptionsInApiFormat.map((optionInApiFormat) => {
+            return {
+              value: optionInApiFormat.value,
+              label: optionInApiFormat.display_name,
+            };
+          })
+        );
         setLoading(false);
       })
       .catch((e: Error) => {
         setLoadingResourceError(e);
         setLoading(false);
       });
```

Comment:
This can be written also in this kind of format:
```js
const fetchData = async (): Promise<void> => {
      try {
        const [apiResource, datePeriod] = await Promise.all([
          api.getResource(resourceId),
          api.getDatePeriod(id),
        ]);
        setResource(apiResource);
        resetForm(datePeriod);
        setIsLoading(false);
      } catch (e) {
        setHasDataLoadingError(e);
        setIsLoading(false);
      }
    };

    fetchData();
```

Reply:
True, I considered it but it creates a new unnecessary function and I just thought plain promise version was cleaner 🤔